source/MaterialXRender/Handlers/ImageHandler.h

```diff
@@ -151,18 +182,22 @@ class ImageHandler
     void addLoader(ImageLoaderPtr loader);
     
     /// Default destructor
-    virtual ~ImageHandler() {}
+    virtual ~ImageHandler() 
+    {
+        delete _restrictions;
```

Comment:
Maybe it would be prettier code to have a virtual method getRestrictions() that returns a pointer to the restrictions for each derived ImageHandler that use this (or null otherwise). To avoid having lifetime management of this member spread over base class and derived classes.